weave/trace/weave_client.py

```diff
@@ -1572,4 +1577,60 @@ def sanitize_object_name(name: str) -> str:
     return res
 
 
+def convert_paths_to_images(obj: str) -> typing.Union[str, PathImage]:
+    """Load or download paths to images as PathImage objects.
+
+    If the path is a local image, open it and return a PathImage object.
+    If the path is a remote image, download it and return a PathImage object.
+    """
+    if is_remote_image(obj) or is_local_image(obj):
+        return PathImage(path=obj, data=None)
+    elif is_base64_image(obj):
+        return PathImage(data=obj, path=None)
+
+    return obj
+
+
+def parse_obj(obj: Any, parsers: list[Callable[..., Any]]) -> Any:
```

Comment:
How much time does this add to the user's process?

Reply:
In my testing, almost zero, except for the remote images, which incurs a network request. I was able to log about ~200 calls per second including multiple small image payloads.